### PR TITLE
Add contract storage indexed requireConfig util

### PR DESCRIPTION
**Pull request by Agusx1211**

### CODE REVIEW

File: contracts/modules/utils/RequireUtils.sol [hunk 1]
```diff
@@ -68,6 +71,51 @@ contract RequireUtils {
...
+      // Store counter factual imageHash
+      initialImageHash[_wallet] = imageHash;
+    }
+
+    // Emit event for easy config retrieval
+    emit RequiredConfig(_wallet, imageHash, _threshold, abi.encode(_members));
+
+    // Store block height for first log
+    if (imageHashBlockHeight[imageHash] == 0) {
+      imageHashBlockHeight[imageHash] = block.number;
```
Comment (PhABC) [inline contracts/modules/utils/RequireUtils.sol:115]: Shouldn't we store the latest blockheight as well? Could be millions of blocks between the first one and the most recent update.

Comment (pkieltyka) [reply contracts/modules/utils/RequireUtils.sol:115]: @PhABC yes agreed. I thought by the design we discussed, this variable would hold the highest block height of the latest image hash update. I notice we also store the initial Image Hash, either that value alone is sufficient for our purposes, or we replace that with initialImageBlockHeight uint256, and we can reconstruct it from the logs

Comment (Agusx1211) [reply contracts/modules/utils/RequireUtils.sol:115]: The process of retrieving a configuration requires two steps:

1) Knowing the imageHash of the wallet
2) Knowing the configuration of the imageHash

For 1) there are two possibilities:

a) the wallet is in counter-factual state or
b) the wallet has been updated

For a) we use the `initialImageHash` mapping, but once the wallet it's updated and it's no longer counter-factual, the imageHash is stored on the contract storage and we can query it just by calling the contract.

And for 2), each imageHash is just a hash of a configuration, so once it's "map" from hash to configuration is created, there is no need to do further updates, either the wallet uses the imageHash and we already have it mapped, or the imageHash changes and we just add another entry to `imageHashBlockHeight`.

**Comment by pkieltyka** [comment]
closing in favour of https://github.com/0xsequence/wallet-contracts/pull/109

**Pull request closed without merging**